lib/services/audio/default_audio_player_service.dart

```diff
@@ -524,6 +525,10 @@ class DefaultAudioPlayerService extends AudioPlayerService {
 
     log.fine('We have completed episode ${_currentEpisode?.title}');
 
+    if (settingsService.deleteDownloadedPlayedEpisodes) {
+      await repository.deleteEpisode(_currentEpisode!);
```

Comment:
ok, I understand :-)

Reply:
I made the correction: now the file is deleted from the storage but the episode and its status remain correctly in the database.
Testing it, listening to the downloaded episode, staying in the downloads screen, I noticed that when the episode is finished, the screen does not refresh automatically (just exit and re-enter and it refreshes).
Analysing the 'delete' button in the episode, I see that the episode is added to a list and then the Bloc architecture (episode_bloc) does the rest.
What would be the best way to achieve both file deletion and screen sync? Should the deletion also be delegated to episode_bloc?

Reply:
Yes, I will need to give this one a little thought. Ideally, the deletion from the service would trigger a reaction further up to cause the screen to refresh. Might need a little refactoring. Will let you know what I find.

Reply:
I woke up at 4am this morning realising what the problem is! Funny how things pop in to your head like that.

In the `episode_bloc` there is an episode listener, but I have applied a filter so it only reacts to downloaded episodes. This no longer applies, as we need to react to updates that include episodes that are no longer downloaded.

In the `_listenEpisodeEvents()` method, if you remove the filter on the listener and then try again. You should find episodes are removed from the Downloads tab as they are removed. This will need a good bit of testing to ensure there are no knock-on effects of removing the filter (especially in terms of performance), but I think this should resolve the issue.

Reply:
I did some tests by removing the filter and it works perfectly. However, I thought that instead of removing the filter, it could be hooked to another change of state, namely played.
When deleting the file, following this feature, the episode will always be played, the filter will pass and the screen will update correctly.
What do you think about?